File: taskmanagerrect/src/types/Task.ts
export enum TaskStatus {
  Pending = 0,
  InProgress = 1,
  Completed = 2
}

export interface Task {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
  isDeleted: boolean;
}

export interface CreateTaskRequest {
  title: string;
  description: string;
  status: TaskStatus;
  isDeleted: boolean;
}

export interface UpdateTaskRequest {
  id: string;
  title: string;
  description: string;
  status: TaskStatus;
}

File: TaskManager/Core/TaskManager.Domain/Entities/Task.cs


namespace TaskManager.Domain.Entities;

public class TaskItem:BaseEntity
{

    public string Title { get; set; } = default!;
    public string Description { get; set; } = default!;
    public TaskStatus Status { get; set; } = TaskStatus.Todo;
    public bool IsDeleted { get; set; } = false;
}
public enum TaskStatus
{
    Todo = 0,
    InProgress = 1,
    Done = 2
}